src/Adapter/CatalogPriceRule/CommandHandler/AddCatalogPriceRuleHandler.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\CatalogPriceRule\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Entity\SpecificPriceRule;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Command\AddCatalogPriceRuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\CommandHandler\AddCatalogPriceRuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\Exception\CatalogPriceRuleException;
+use PrestaShop\PrestaShop\Core\Domain\CatalogPriceRule\ValueObject\CatalogPriceRuleId;
+use PrestaShopException;
+
+/**
+ * Handles adding new catalog price rule using legacy object model
+ */
+final class AddCatalogPriceRuleHandler implements AddCatalogPriceRuleHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(AddCatalogPriceRuleCommand $command)
```

Comment:
@matks, I found this smartyCache cleaning unnecessary, but I'm not really sure, wdyt?
https://github.com/PrestaShop/PrestaShop/blob/0181a42b0a5f611a32949f5ba45d199ab85cfcc8/controllers/admin/AdminSpecificPriceRuleController.php#L376

Reply:
You can read about that here : https://github.com/PrestaShop/PrestaShop/issues/13627#issuecomment-492615902

In fact cache need to be cleared in order to update price in all existing smarty cache but doing here is not really good as you can read on my report.

Reply:
Very interesting, thanks @Matt75 !
@zuk3975 I suggest we remove it for now but create a dedicated issue for that.

I can imagine 3 solutions for this issue (there might be more):
- display a warning message after a catalog price rule has been created "eyh ! you created a rule, you probably want to clear the cache to make sure FO display is right !" with a link/button to perform the clear (additionnaly: using an ajax request so we dont freeze the screen ❤️)
- put a timer to clear cache 1 minute after last catalog price rule modification in order to avoid to perform 1 clear by rule created
- allow merchant to control whether he wants to clear the cache manually or automatically with a boolean configuration value

Reply:
created a reminder issue for that https://github.com/PrestaShop/PrestaShop/issues/14761